### PR TITLE
v2-handle-stuck-job: Handle stale or stuck jobs

### PR DESCRIPTION
**Pull request by varunsharma0286**

### CODE REVIEW

File: server/http.go [hunk 1]
```diff
@@ -88,7 +91,15 @@ func runSecretScanStandalone(writer http.ResponseWriter, request *http.Request)
 		return
 	}
 	fmt.Printf("Secret Scan triggered for %s: ", req.ImageNameWithTag)
-	res, err := scan.ExtractAndScanImage(req.ImageNameWithTag)
+
+	scanCtx := scan.NewScanContext("")
```
Comment (noboruma) [inline server/http.go:95]: Open to discussion but I am not 100% sure we want status update for operations happening outside gRPC
The status update is only needed for console/gRPC

Comment (varunsharma0286) [reply server/http.go]: Agree... we can work on the non-grpc workflow later.

**Review by noboruma — APPROVED** [review]

File: scan/scan_context.go [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 	ScanID         string
-	Aborted        bool
+	Aborted        atomic.Bool
 	ScanStatusChan chan bool
 }
 
 func NewScanContext(scanID string) *ScanContext {
 	statusChan := make(chan bool)
-	return &ScanContext{scanID, false, statusChan}
+	obj := ScanContext{scanID, atomic.Bool{}, statusChan}
```
Comment (noboruma) [inline scan/scan_context.go:13]: nit: `atomic.Bool{}` is already initialized as `false`

File: scan/process_image.go [hunk 3]
```diff
@@ -821,7 +821,7 @@ func ExtractAndScanFromTar(tarFolder string, imageName string) (*ImageExtraction
 
 func CheckScanStatus(scanCtx *ScanContext) error {
 	if scanCtx != nil {
-		if scanCtx.Aborted == true {
+		if scanCtx.Aborted.Load() == true {
```
Comment (noboruma) [inline scan/process_image.go:824]: nit: `if scanCtx.Aborted.Load() {`

**Review by noboruma — APPROVED** [review]

**Pull request merged**